rir/src/compiler/analysis/scope.cpp

```diff
@@ -138,15 +138,18 @@ AbstractResult ScopeAnalysis::apply(ScopeAnalysisState& state,
         auto force = Force::Cast(i);
         auto arg = force->arg<0>().val();
         if (!arg->type.maybeLazy()) {
-            effect.max(state.returnValues[i].merge(ValOrig(arg, i, depth)));
+            if (!arg->type.maybePromiseWrapped())
```

Comment:
Only `!maybePromiseWrapped` is enough

Reply:
the second statement is executed for both